### PR TITLE
Combine prflx and stats discussion in a specific Statistics section

### PR DESCRIPTION
**Pull request by juberti**
Followup from #11, as discussed today

### CODE REVIEW

**Comment by juberti** [comment]
Will wait for LGTM from youennf@ before merging.

**Comment by juberti** [comment]
Rethought my approach here and rewrote the section accordingly. PTAL.

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 1]
```diff
@@ -155,36 +157,38 @@ An ICE agent that supports mDNS candidates MUST support the situation where the
...
+
+The goal of this mechanism is to keep knowledge of host IP addresses
+within the ICE agent while continuing to allow the application to
+transmit ICE candidates. Besides keeping host IP addresses out of 
+ICE candidates, implementations must take steps to prevent host IP 
+addresses from being exposed to web applications through other means.
+
+Statistics related to ICE candidates that are accessible to the web
+application MUST NOT contain the IP address of a local or remote mDNS
+mDNS candidate; the mDNS name SHOULD be used instead.
```
Comment (tQsW) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]:
```suggestion
candidate; the mDNS name SHOULD be used instead.
```

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 2]
```diff
@@ -155,36 +157,38 @@ An ICE agent that supports mDNS candidates MUST support the situation where the
...
+
+In addition, a peer-reflexive remote candidate may be constructed 
+from a remote host IP address as a result of an ICE connectivity
+check, as described in Section 7.3.1.3 of {{RFC8445}}. This check
+may arrive before the candidate due to signaling or mDNS
+resolution delays. 
+
+To prevent disclosure of the host IP address to the application in
+this scenario, statistics related to ICE candidates MUST NOT 
+contain the the IP address of any peer-reflexive candidate, unless that IP
```
Comment (tQsW) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]:
```suggestion
contain the IP address of any peer-reflexive candidate, unless that IP
```

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 3]
```diff
@@ -155,36 +157,38 @@ An ICE agent that supports mDNS candidates MUST support the situation where the
...
-APIs leaking IP addresses
-----------------------------
-
-When there is no user consent, the following filtering should be done to prevent private IP address leakage:
-
-1. ICE candidates with an IP address are not exposed as ICE candidate events.
-
-2. Server reflexive ICE candidate raddr field is set to 0.0.0.0 and rport to 0.
-
-3. SDP does not expose any a=candidate line corresponding to an ICE candidate which contains an IP address.
```
Comment (tQsW) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: This item need to be rephrased to limit the exposure of IP for mDNS candidates. I think the meaning has changed after we changed the original "ICE host candidate" to "ICE candidate." This item is also not in the revision of this PR. Though I feel it should be part of the definition of mDNS candidate (and its signaling), do we also want to keep it in {#privacy}?

Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: I think this should be covered by the #gathering section - thoughts? Or do we want to reiterate this for getLocalDescription/getRemoteDescription?

Comment (tQsW) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: > I think this should be covered by the #gathering section

Agreed, it reads to me now step 6 in the #gathering section should have covered it. It is not specific to SDP or local description, but SDP is rather an example protocol of exchange in 8445 and local description reads to me more WebRTCSpec. I am adding a few words to refer to the candidate exchange in 8445. Let me know if it reads okay.

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 4]
```diff
@@ -122,7 +122,8 @@ For any host candidate gathered by an ICE agent as part of {{RFC8445}} section 5
 
 5. Store the mDNS hostname and its related IP address in the ICE agent for future reuse.
 
-6. Replace the IP address of the ICE candidate with its mDNS hostname, and expose the candidate as usual.
+6. Replace the IP address of the ICE candidate with its mDNS hostname, expose
```
Comment (tQsW) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: I didn't find a way to suggest a change in a non-diff area other than creating another commit. Also used the uppercase initial to be consistent. PTAL.

Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: OK, LG.

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 3 above)
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: Thought about it a bit more, and not sure citing 8445 makes sense here. We're not actually sending the candidate to the remote side, as noted in 8445, we're just giving it to the web app here.

Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: As a result, I left step 6 mostly as-is.

Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: OK, will add a note on this.

Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: Added some new text, PTAL.

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 5]
```diff
@@ -138,9 +138,15 @@ An implementation may also detect that mDNS is not supported by the available ne
 The ICE agent may skip steps 2 and 3 and directly decide to not expose the host candidate.
 
 This procedure ensures that a mDNS name is used to replace only one IP address.
-Specifically, an ICE agent using an interface with both IPv4 and IPv6 addresses MUST expose a different mDNS name for each address.
+Specifically, an ICE agent using an interface with both IPv4 and IPv6 addresses MUST 
+expose a different mDNS name for each address.
 
-Any server-reflexive candidates generated from a mDNS local candidate MUST have their raddr field set to 0.0.0.0 and their rport field set to 0.
+Any server-reflexive candidates generated from a mDNS local candidate MUST have
```
Comment (tQsW) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: nit: a mDNS or an mDNS?

Comment (tQsW) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: nit: a mDNS or an mDNS?

Comment (tQsW) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: nit: a mDNS or an mDNS?

Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: an mDNS, will change here and elsewhere in followup

**Comment by youennf** [comment]
LGTM

**Pull request merged**